[jit] Added an API to remove list mutations and replace with variadic cat until fixed point

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61498
* #60909
* #60908
* **#60776**
* #60775
* #60774

Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

navahgar committed:
[jit] Added an API to remove list mutations and replace with variadic cat until fixed point

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

eellison commented on torch/csrc/jit/passes/concat_opt.cpp:
```diff
@@ -590,5 +591,16 @@ bool UseVariadicCat(const std::shared_ptr<Graph>& graph) {
   return changed;
 }
 
+bool RemoveListMutationAndUseVariadicCat(const std::shared_ptr<Graph>& graph) {
```
Looks good but as discussed offline eventually we would like to make this one pass instead of running each back and forth

navahgar replied:
Sure. That requires refactoring some of the code to work on nodes instead of graphs on the whole. Will do that as a follow-up.

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).

navahgar committed:
Update on "[jit] Added an API to remove list mutations and replace with variadic cat until fixed point"


Differential Revision: [D29406099](https://our.internmc.facebook.com/intern/diff/D29406099)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29406099).
